src/VirtoCommerce.Platform.Web/SignalR/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,58 @@
+
+using System;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Azure.SignalR;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using VirtoCommerce.Platform.Core.SignalR;
+
+namespace VirtoCommerce.Platform.Web.SignalR
+{
+    public static class ServiceCollectionExtensions
+    {
+        private static readonly SignalROptions _signalROptions = new SignalROptions();
+
+        public static void AddSignalR(this IServiceCollection services, IConfiguration configuration)
+        {
+            var signalRSection = configuration.GetSection("SignalR");
+            signalRSection.Bind(_signalROptions);
+
+            services.AddOptions<SignalROptions>().Bind(configuration.GetSection("SignalR")).ValidateDataAnnotations();
+
+            var signalRServiceBuilder = services.AddSignalR().AddNewtonsoftJsonProtocol();
+
+            // SignalR scalability configuration. RedisBackplane (default provider) will be activated only when RedisConnectionString is set
+            // otherwise no any SignalR scaling options will be used
+            if (_signalROptions.ScalabilityProvider == SignalRScalabilityProvider.AzureSignalRService)
+            {
+                signalRServiceBuilder.AddAzureSignalR(configuration);
+            }
+            else
+            {
+                signalRServiceBuilder.AddRedisBackplane(configuration);
+            }
+        }
+
+        public static void AddAzureSignalR(this ISignalRServerBuilder builder, IConfiguration configuration)
+        {
+            var azureSignalRConnectionString = _signalROptions.AzureSignalRService.ConnectionString;
+            if (string.IsNullOrEmpty(azureSignalRConnectionString))
+            {
+                throw new InvalidOperationException("SignalR:AzureSignalRService:ConnectionString  must be set");
+            }
+            builder.AddAzureSignalR(options =>
+            {
+                options.Endpoints = new [] { new ServiceEndpoint(azureSignalRConnectionString) };
+            });
+        }
+
+        public static void AddRedisBackplane(this ISignalRServerBuilder builder, IConfiguration configuration)
```

Comment:
This registration can be moved from here to the Redis `serviceCollectionExtensions`.

Reply:
Fixed